esp/GBPlay/main/tasks/socket_manager.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _SOCKET_MANAGER_H
+#define _SOCKET_MANAGER_H
+
+extern int sock;
+extern char rx_buffer[16];
```

Comment:
A better way to do this is probably using a [FreeRTOS queue](https://www.freertos.org/Embedded-RTOS-Queues.html) or emitting an [ESP event](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/esp_event.html).

Then thread safety/blocking/etc would be taken care of for us, which would simplify things for when we have the GB hooked up and another higher-level task acting as the relay between GB <-> TCP.

The manager could be something like (pseudocode):
```
loop forever
    packet = recv()

    // Using FreeRTOS queue
    enqueue(response_queue, packet)

    // Alternately, using ESP event
    emit(EVENT_GOT_DATA, packet);
```
The application layer code would then either block on the response queue or wait for the "got data" event. Of the two, I feel like the event-based approach makes more sense (basically treating receiving data similarly to an interrupt).

What do you think? What's here works well enough for now but I think we should refactor once we're at the point of hooking up the GB to the TCP connection - for simplicity of task interaction and so we only use the CPU when absolutely necessary.

Reply:
yeah my original plan was to implement a socket consumer queue to handle this, it'll definitely work for now but absolutely a future improvement. I agree with the event style